### PR TITLE
multicluster dns debugging and fix

### PR DESCRIPTION
**Pull request by stevenctl**
For debugging/fixing #27342 

Link to the tests we care about here: https://prow.istio.io/?pull=27343&job=integ-pilot-multicluster-tests_istio

### CODE REVIEW

**Review by ramaraochavali — APPROVED** [review]
Let us try

File: pkg/test/kube/dump.go [hunk 1]
```diff
@@ -42,6 +42,27 @@ func outputPath(workDir string, cluster resource.Cluster, pod kubeApiCore.Pod, n
...
 
+func DumpDebug(a resource.Cluster, workDir, systemNamespace, endpoint string) {
+	errFmt := "failed dumping " + endpoint + " for " + a.Name() + ": %s: %v"
+	pods, err := a.PodsForSelector(context.TODO(), systemNamespace, "istio=pilot")
+	if err == nil && len(pods.Items) == 0 {
+		err = fmt.Errorf("0 pods found for istio: pilot in %q", systemNamespace)
+	}
+	if err == nil {
+		name := pods.Items[0].Name
+		out, _, err := a.PodExec(name, systemNamespace, "discovery", "curl 127.0.0.1:15000"+endpoint)
```
Comment (howardjohn) [inline pkg/test/kube/dump.go:53]: This will break distroless, we could instead do `pilot-discovery request GET endpoint` I think

**Review by nmittler — APPROVED** [review]

File: tests/integration/pilot/routing_test.go [hunk 2]
```diff
@@ -19,14 +19,14 @@ import (
...
 )
 
 func TestTraffic(t *testing.T) {
 	framework.
 		NewTest(t).
 		Features("traffic.routing", "traffic.reachability", "traffic.shifting").
 		Run(func(ctx framework.TestContext) {
-			common.RunTrafficTest(ctx, apps)
+			ctx.FailNow()
+			//common.RunTrafficTest(ctx, apps)
```
Comment (hzxuzhonghu) [inline tests/integration/pilot/routing_test.go:30]: Just skip the test?

Comment (stevenctl) [reply tests/integration/pilot/routing_test.go:30]: The point of this PR is to dump debugging info. We don't store artifacts in CI unless we fail.

**Pull request merged**